### PR TITLE
Nan 2

### PR DESCRIPTION
**Pull request by paulcbetts**
This PR moves us to nan 2.0 which is required for Chrome 44+

## TODO:

- [x] Figure out what to do about `Reset` call

### CODE REVIEW

File: src/common.cc [hunk 1]
```diff
@@ -106,60 +106,60 @@ void PostEventAndWait(EVENT_TYPE type,
...
+  Nan::HandleScope scope;
 
-  if (!args[0]->IsFunction())
-    return NanThrowTypeError("Function required");
+  if (!info[0]->IsFunction())
+    return Nan::ThrowTypeError("Function required");
 
-  NanAssignPersistent(g_callback, Local<Function>::Cast(args[0]));
-  NanReturnUndefined();
+  g_callback.Reset( Local<Function>::Cast(info[0]));
```
Comment (zcbenz) [inline src/common.cc]: You need to pass isolate for Reset:

```c++
g_callback.Reset(info.GetIsolate(), Local<Function>::Cast(info[0]));
```

File: src/common.cc [hunk 2]
```diff
@@ -106,60 +106,60 @@ void PostEventAndWait(EVENT_TYPE type,
...
+  Nan::HandleScope scope;
 
-  if (!args[0]->IsFunction())
-    return NanThrowTypeError("Function required");
+  if (!info[0]->IsFunction())
+    return Nan::ThrowTypeError("Function required");
 
-  NanAssignPersistent(g_callback, Local<Function>::Cast(args[0]));
-  NanReturnUndefined();
+  g_callback.Reset(v8::Isolate::GetCurrent(), Local<Function>::Cast(info[0]));
```
Comment (paulcbetts) [inline src/common.cc]: @zcbenz Is this the same? Tests seem to pass

Comment (zcbenz) [inline src/common.cc]: Looks good to me.

**Comment by ChALkeR** [comment]
Adding to the list: https://github.com/nodejs/node/issues/2798.
Also, I'm pretty sure that the latest published version still depends on nan-1.6.1.

File: src/common.cc (hunk 2 above)
Comment (kkoopa) [inline src/common.cc]: `g_callback` should be a `Nan::Persistent`, which does not take an `Isolate *`.

Comment (paulcbetts) [inline src/common.cc]: hwhoops, thanks @kkoopa, will fix that up

**Pull request merged**